src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java

```diff
@@ -0,0 +1,152 @@
+package com.basho.riak.client.core.util;
+
+import org.junit.Assume;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertEquals;
+
+public class DefaultCharsetTest
+{
+    @Test
+    public void testDefaultIfNoPropertyDefined()
+    {
+        Charset systemDefault = Charset.defaultCharset();
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean noCharsetPropertyDefined = definedCharsetName == null;
+        Assume.assumeTrue(noCharsetPropertyDefined);
+
+        assertEquals(systemDefault, clientDefault);
+    }
+
+    @Test
+    public void testDefinedCharset()
+    {
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean charsetPropertyDefined = definedCharsetName != null && !definedCharsetName.isEmpty();
+
+        Assume.assumeTrue(charsetPropertyDefined);
+
+        Charset definedCharset = Charset.forName(definedCharsetName);
+        assertEquals(definedCharset, clientDefault);
+    }
+
+    @Test
+    public void testStaticInitializerValidDefinedCharset()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException
+    {
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+        Charset systemDefault = Charset.defaultCharset();
```

Comment:
This var (systemDefault) might be refactored to class final static protected attribute and reused after that whenever it needs. IMO  It will increase readability.